extensions/jenkins/src/jenkins.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  ActionPanel,
+  Action,
+  showToast,
+  Toast,
+  List,
+  Alert,
+  confirmAlert,
+  Icon,
+  useNavigation,
+  Form,
+} from "@raycast/api";
+import { Search } from "./search";
+import { Jenkins, JenkinsAPI } from "./lib/api";
+import { addJenkins, deleteJenkins, listJenkins } from "./lib/storage";
+import { useState, useCallback, useEffect } from "react";
+
+export default function Command() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [jenkinsList, setJenkinsList] = useState<Jenkins[]>([]);
+
+  const search = useCallback(
+    async function search(searchText: string) {
+      setIsLoading(true);
+      try {
+        const jenkinsList = await listJenkins();
+        const results = jenkinsList.filter((j) => j.name.toLowerCase().includes(searchText.toLowerCase()));
+        setJenkinsList(results);
+      } catch (err) {
+        showToast({ style: Toast.Style.Failure, title: "Search Failed", message: String(err) });
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [setIsLoading, setJenkinsList]
+  );
+
+  useEffect(() => {
+    search("");
+  }, []);
+
+  return (
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={search}
+      searchBarPlaceholder="Search jenkins..."
+      throttle
+      actions={
+        <ActionPanel>
+          <Action.Push
+            icon={Icon.NewDocument}
+            title="Add jenkins"
+            target={<AddJenkins setJenkinsList={setJenkinsList} />}
+          />
+        </ActionPanel>
```

Comment:
The confusing part 🙂

Reply:
In my mind, Preferences can only be used to set one account information, but this extension is being to manage multi jenkins acounts 

Reply:
Because it could not add jenkins with empty list, this is why I add actions at List 